### ISSUE TITLE
Vagrant does not reference current VM on empty VM hash

### ISSUE THREAD

**Issue by codekandis** [hashicorp/vagrant#12179]
I'm using some functions tho start, provision, halt and destroy my VMs with vagrant. I wrote an CLI completion to reference several VMs by their hashes. If not provided the VM in the current directory should be used instead.

```
machineId="$( someFunctionReturningVmHashOrEmptyString )"
vagrant destroy "${machineId}"
```

If no machine ID is provided the script leads to a call like `vagrant destroy ""`. Vagrant doesn't validate the empty argument string and takes one of the known VMs instead (the first of all known machines sorted by their hashes, I guess).

This references all calls of vagrant referencing a VM.

### Vagrant version

2.2.14

### Host operating system

Linux hostname 4.19.0-14-amd64 #1 SMP Debian 4.19.171-2 (2021-01-30) x86_64 GNU/Linux

### Expected behavior

The current machine must be referenced if an empty string has been passed.

### Actual behavior

One of the globally known VMs will be referenced if an empty string has been passed.

### Steps to reproduce

1. create multiple VMs
2. cd into the directory of one VM
3. call `vagrant destroy ""` or any other vagrant command referencing VMs

**Comment by soapy1** [comment]
Heya, thanks for opening up an issue. https://github.com/hashicorp/vagrant/pull/12188 fixes this. However, I think the expected behavior for a `vagrant destroy ""` is to kick back an error saying that the machine `""` has not been found.

**Comment by codekandis** [comment]
In fact quotes are meant to include multiple words as one argument in the first place (special chars, etc., as well). So empty quotes are an empty string. So in fact `vagrant destroy ""` is equal to `vagrant destroy` and references a possible machine in the current working directory and should throw an error if no machine has been found.

**Comment by soapy1** [comment]
> In fact quotes are meant to include multiple words as one argument in the first place (special chars, etc., as well). So empty quotes are an empty string.

👍 

> So in fact `vagrant destroy ""` is equal to `vagrant destroy`

I don't think this is necessarily true. For example, an empty string is not the same as `nil`. I think in the case of `vagrant destroy <something>`, Vagrant should try to understand what `<something>` is. If `<something>` is `nil`, then the intention is to destroy all the machines in the current environment. If Vagrant is unable to interpret the argument, then it should be expressing that to the user in the form of an error or warning. 

So, the following Vagrantfile is valid (albeit confusing) 

```
Vagrant.configure("2") do |config|
  config.vm.define "" do |c|
    c.vm.box = "hashicorp/bionic64"
  end

  config.vm.define "my-machine" do |c|
    c.vm.box = "hashicorp/bionic64"
  end
end
```

With this Vagrantfile `vagrant destroy ""` should only destroy the machine `""`

**Comment by codekandis** [comment]
Ok, this leads to some new discoveries.

I tried to manage VMs by its name in the global scope. And vagrant does not manage VMs by its name outside of the working directory of the Vagrantfile. If I would call `vagrant destroy my-machine` vagrant would return the `The machine with the name 'my-machine' was not found configured for this Vagrant environment.`

Thats why I wrote a CLI completion script translating a machine name to its hash, which in the end could lead to an empty hash string.

I do not have multiple machines managed by one Vagrantfile. So I never tried to manage a VM by its name in the same working directory as the Vagrantfile. I didn't know I could reference a VM by its name in the working directory. I'd have always used hashes instead.

I didn't knew a machine can have an empty name. This, for me, looks kind of unusual. But in this case I uinderstand the different handling of an empty machine reference.
